docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -122,6 +122,43 @@ Elliptic Curve Signature Algorithms
         :returns: A new instance of a :class:`EllipticCurvePublicKey`
             provider.
 
+Elliptic Curve Key Exchange algorithm
+-------------------------------------
+
+.. class:: ECDH(algorithm)
```

Comment:
Another option would be something like:
keyex = ECDH()
shared_key = keyex.exchange(private_key, peer_public_key, KDF(..))

If private_key is None we generate one on the spot
the keyex.private_key will return you the private key should you need it ?